src/viking.js

```diff
@@ -1,11 +1,96 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage; //should remove the received damage from the health property
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(theDamage) {
+    this.health -= theDamage;
+    if (this.health > 0) {
+      return `${this.name} has received ${theDamage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(theDamage) {
+    this.health -= theDamage;
+    if (this.health > 0) {
+      return `A Saxon has received ${theDamage} points of damage`;
+    } else {
+      return "A Saxon has died in combat";
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  vikingArmy = [];
+  saxonArmy = [];
+
+  addViking(viking) {
+    this.vikingArmy.push(viking); // to add
+  }
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon);
+  }
+  vikingAttack() {
+    let randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    let result = this.saxonArmy[randomSaxon].receiveDamage(
+      this.vikingArmy[randomViking].strength
+    );
+
+    if (this.saxonArmy[randomSaxon].health <= 0) {
+      this.saxonArmy.pop(randomSaxon);
+    }
+    return result;
+  }
+  saxonAttack() {
+    let randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    let result = this.vikingArmy[randomViking].receiveDamage(
+      this.saxonArmy[randomSaxon].strength
+    );
+
+    if (this.vikingArmy[randomViking].health <= 0) {
+      this.vikingArmy.pop(randomViking);
+    }
+    return result;
+  }
+  showStatus() {
+    if (this.saxonArmy == 0) {
```

Comment:
Always remember to use the three equal signs `===` instead of just `==`. In this case it's ok, but can lead to mistakes and/or bugs later in the code.